demos/common/python/openvino/model_zoo/model_api/adapters/openvino_adapter.py

```diff
@@ -0,0 +1,133 @@
+"""
+ Copyright (c) 2021 Intel Corporation
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+
+import logging as log
+from collections import deque
+
+try:
+    from openvino.inference_engine import IECore, get_version
+    openvino_absent = False
+except ImportError:
+    openvino_absent = True
+
+import ngraph
+
+from .model_adapter import ModelAdapter, Metadata
+from ..pipelines import parse_devices
+
+
+def create_core():
+    if openvino_absent:
+        raise ImportError('The OpenVINO package is not installed')
+
+    log.info('OpenVINO Inference Engine')
+    log.info('\tbuild: {}'.format(get_version()))
+    return IECore()
```

Comment:
So you are hiding all openvino calls. That feels like an attempt to kill openvino with your craft. We should promote openvino, not fully replace.

Even before introduction of adapters ppl had caserns that now there will be two APIs. The patch makes this concern reality.

@vladimir-dudnik, I think this is where you need to consult with the architecture team. CC @yury-gorbachev

Reply:
In modelAPI demos the Openvino calls have been already hidden in `AsyncPipeline` class, and in the demo we only create `IECore()`. Does the hiding of Openvino import and core creation prevents the promotion of Openvino in demos at all? 

The idea to abstract the executor and its specific calls implied that for user the details of any adapter implementation will be not interested and therefore separated from demos.

Reply:
`AsyncPipeline` never assumed openvino can be absent.

> Does the hiding of Openvino import and core creation prevents the promotion of Openvino in demos at all?

I guess so. You've proved it yourself with

> for user the details of any adapter implementation will be not interested

But it's up to bigger people to decide. For now it's just yet another concern.

Reply:
So this is not a problem